### PR TITLE
Add API's and fields available in Genie 3.1.0

### PR DESCRIPTION
**Pull request by tgianos**

### CODE REVIEW

**Comment by tgianos** [comment]
@charsmith @irontable @stephen-mw These are what I think are the changes needed to they Python client for the upcoming [Genie 3.1.0](https://github.com/Netflix/genie/milestone/10) release. I'm not sure if this pygenie repo is yet the active one but figured best to open it here. Don't see a develop branch or anything so just set merge to master and we can merge it when needed

File: pygenie/jobs/core.py [hunk 1]
```diff
@@ -203,6 +204,15 @@ def _add_dependency(self, dep):
         if dep not in self._dependencies:
             self._dependencies.append(dep)
 
+    def _add_config(self, config):
```
Comment (mprimi) [inline pygenie/jobs/core.py]: Does this automagically work for both a single config (i.e. argument type string) and a collection of them (argument type list)?

**Comment by mprimi** [comment]
Also, no extra tests for the new APIs?
(I'm looking at this repo now for the first time, maybe there are no tests?)

File: pygenie/client.py [hunk 2]
```diff
@@ -720,6 +719,250 @@ def get_commands(self, filters=None, req_size=1000):
...
+        Args:
+            command_id (str): the command id to remove configs
+
+        Returns: None
+
+        Example:
+            >>> remove_all_command_configs('command2')
+
+        """
+        path = self.path_command + '/' + command_id + '/configs'
```
Comment (charsmith) [inline pygenie/client.py]: Just as a general note this type of code is typically written like:

```
'/'.join((self.path_command, command_id, 'configs'))
```

or 

```
os.sep.join((self.path_command, command_id, 'configs'))
```

File: pygenie/jobs/core.py (hunk 1 above)
Comment (tgianos) [inline pygenie/jobs/core.py]: looks like... python voodo... someone better at python like @charsmith or @irontable would be better to answer. I just copied what was done for job dependencies.

**Comment by tgianos** [comment]
@irontable I rebased this on top of your last two commits (d31966ec539032a58421a5099dfdec89021d51e7 and b6e5cae2567a6a69c3346defec26b452bf31256f) which said they were only "bump version" in the commit message and now the PR build is broken any idea how that could happen? Looking at the commits they indeed look only like version bumps.

**Comment by Rustem** [comment]
Hey guys, looks like you did not pass configs when building payload

**Pull request merged**